src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction.cs

```diff
@@ -476,13 +476,36 @@ private SyntaxNode CreateThroughExpression(SyntaxGenerator factory)
                     // uncommon case and optimize for the common one - in other words, we only apply the cast
                     // in cases where we can unambiguously figure out which interface we are trying to implement.
                     var interfaceBeingImplemented = State.InterfaceTypes.SingleOrDefault();
-                    if ((interfaceBeingImplemented != null) && (!throughMemberType.Equals(interfaceBeingImplemented)))
+                    if ((interfaceBeingImplemented != null))
                     {
-                        through = factory.CastExpression(interfaceBeingImplemented,
-                            through.WithAdditionalAnnotations(Simplifier.Annotation));
+                        if (!throughMemberType.Equals(interfaceBeingImplemented))
+                        {
+                            through = factory.CastExpression(interfaceBeingImplemented,
+                                through.WithAdditionalAnnotations(Simplifier.Annotation));
```

Comment:
Since this code [was already here](https://github.com/dotnet/roslyn/blob/85ca862c20fd1cb4192f3a42521eeadcba6828cb/src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction.cs#L482), yes, I believe it works. :)  My understanding is it simplifies things like `(global::IA)` to `(IA)`.